core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/configuration/script/ScriptConfiguration.java

```diff
@@ -384,4 +488,26 @@ private String getInsertStatement(Script script) {
         }
     }
 
+    public void restoreDeletedScript(Script script) {
+        log.trace(MessageFormat.format("Restoring Deleted script {0}", script.getMetadataKey().toString()));
+        ScriptVersionConfiguration.getInstance().restoreDeletedScriptVersion(script.getVersion());
+        getScriptRestoreStatement(script).
+                ifPresent(getMetadataRepository()::executeUpdate);
+    }
+
+    private Optional<String> getScriptRestoreStatement(Script script) {
+        if (!existsById(script.getMetadataKey().getScriptId())){
+            CachedRowSet crsScript = getMetadataRepository().executeQuery(
+                    String.format(EXISTS_DELETED_BY_ID_QUERY,
+                            SQLTools.getStringForSQL(script.getMetadataKey().getScriptId()),
+                            SQLTools.getStringForSQL(script.getDeletedAt())), "reader");
+            if (crsScript.size() != 0){
+                return Optional.of(String.format(
+                        RESTORE_SOFT_DELETED_BY_ID_QUERY,
+                        SQLTools.getStringForSQL(script.getMetadataKey().getScriptId()),
+                        SQLTools.getStringForSQL(script.getDeletedAt())));
+            }
+        }
+        return Optional.empty();
+    }
```

Comment:
restoreDeletedScript should receive the ScriptKey containing scriptId, scriptVersion & deletedAt.